src/freenet/support/SectoredRandomGrabArray.java

```diff
@@ -173,7 +173,8 @@ private RandomGrabArrayItem removeRandomLimited(
 				Logger.error(this, "Slot "+x+" is null for client "+grabClients[x]);
 				excluded++;
 				if(excluded > MAX_EXCLUDED) {
-					Logger.normal(this, "Too many sub-arrays are entirely excluded on "+this+" length = "+grabArrays.length, new Exception("error"));
+					if(logMINOR)
```

Comment:
Please stop doing this. Optimizations of this kind haven’t been necessary for over 10 years now.

Reply:
This was observed being run 100k times — while 100 times creating the backtrace is OK, 100k backtraces (`new Exception("error")`) actually cause massive load.

The reason why if(logMINOR) is useful here is not that it prevets running Logger.minor(…), but that it prevents the creation of the backtrace which will be run before Logger.minor(…) decides whether to log.

Reply:
I know exactly what that line does. 🙂

Nevertheless, creating 100000 exceptions takes less than 36 ms on my MacBook. I’d much rather simply get rid of creating the exception than littering the source code with those `if` blocks.

And apparently that exception doesn’t even help because otherwise you’d simply fix the underlying issue, right? 😁